src/connectors/toggle-refinement/connectToggleRefinement.ts

```diff
@@ -73,95 +146,66 @@ const createSendEvent = ({ instantSearchInstance, attribute, on, helper }) => (
  *  - switch between two values.
- *
- * @type {Connector}
- * @param {function(ToggleRenderingOptions, boolean)} renderFn Rendering function for the custom **Toggle** widget.
- * @param {function} unmountFn Unmount function called when the widget is disposed.
- * @return {function(CustomToggleWidgetParams)} Re-usable widget factory for a custom **Toggle** widget.
- * @example
- * // custom `renderFn` to render the custom ClearAll widget
- * function renderFn(ToggleRenderingOptions, isFirstRendering) {
- *   ToggleRenderingOptions.widgetParams.containerNode
- *     .find('a')
- *     .off('click');
- *
- *   var buttonHTML = `
- *     <a href="${ToggleRenderingOptions.createURL()}">
- *       <input
- *         type="checkbox"
- *         value="${ToggleRenderingOptions.value.name}"
- *         ${ToggleRenderingOptions.value.isRefined ? 'checked' : ''}
- *       />
- *       ${ToggleRenderingOptions.value.name} (${ToggleRenderingOptions.value.count})
- *     </a>
- *   `;
- *
- *   ToggleRenderingOptions.widgetParams.containerNode.html(buttonHTML);
- *   ToggleRenderingOptions.widgetParams.containerNode
- *     .find('a')
- *     .on('click', function(event) {
- *       event.preventDefault();
- *       event.stopPropagation();
- *
- *       ToggleRenderingOptions.refine(ToggleRenderingOptions.value);
- *     });
- * }
- *
- * // connect `renderFn` to Toggle logic
- * var customToggle = instantsearch.connectors.connectToggleRefinement(renderFn);
- *
- * // mount widget on the page
- * search.addWidgets([
- *   customToggle({
- *     containerNode: $('#custom-toggle-container'),
- *     attribute: 'free_shipping',
- *   })
- * ]);
  */
-export default function connectToggleRefinement(renderFn, unmountFn = noop) {
+const connectToggleRefinement: ToggleRefinementConnector = function connectToggleRefinement(
+  renderFn,
+  unmountFn = noop
+) {
   checkRendering(renderFn, withUsage());
 
-  return (widgetParams = {}) => {
-    const { attribute, on: userOn = true, off: userOff } = widgetParams;
+  return widgetParams => {
+    const { attribute, on: userOn = true, off: userOff } = widgetParams || {};
 
     if (!attribute) {
       throw new Error(withUsage('The `attribute` option is required.'));
     }
 
     const hasAnOffValue = userOff !== undefined;
-    const hasAnOnValue = userOn !== undefined;
-    const on = hasAnOnValue ? toArray(userOn).map(escapeRefinement) : undefined;
+    const on = toArray(userOn).map(escapeRefinement);
     const off = hasAnOffValue
       ? toArray(userOff).map(escapeRefinement)
       : undefined;
 
-    let sendEvent;
+    let sendEvent: SendEventForToggle;
 
-    const toggleRefinementFactory = helper => ({ isRefined } = {}) => {
-      // Checking
+    const toggleRefinementFactory = (helper: AlgoliaSearchHelper) => (
+      {
+        isRefined,
+      }: {
+        isRefined: boolean;
+      } = { isRefined: false }
+    ) => {
       if (!isRefined) {
         sendEvent('click', isRefined);
```

Comment:
I'm not super familiar with this but isn't we supposed to pass the `attribute:{{value}}`  here ?

Reply:
let's not change it in this PR :)